### ISSUE TITLE
Plugin crashes on background for flutter 2.x

### ISSUE THREAD

**Issue by hlavki** [transistorsoft/flutter_background_geolocation#537]
## Your Environment
* Plugin version: 4.0.0-nullsafety.0
* Platform: Android
* OS version: Android 10
* Device manufacturer / model: Samsung S9
* Flutter info (`flutter doctor`):
    ```
    Doctor summary (to see all details, run flutter doctor -v):
    [✓] Flutter (Channel stable, 2.0.3, on Linux, locale sk_SK.UTF-8)
    [✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    [✓] Chrome - develop for the web
    [!] Android Studio (not installed)
    [✓] VS Code (version 1.54.3)
    [✓] Connected device (2 available)

    ! Doctor found issues in 1 category.
    ````
* Plugin config:
```dart
    bg.BackgroundGeolocation.ready(bg.Config(
      url: MoresConfig.userLocationUrl,
      maxBatchSize: 5,
      extras: {"instanceId": _instanceInfo.id},
      maxDaysToPersist: 3,
      debug: false,
      autoSync: true,
      triggerActivities: 'on_foot, walking, running',
      desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
      notificationChannelName: strings.notificationChanel,
      stopOnTerminate: false,
      allowIdenticalLocations: false,
      startOnBoot: true,
      enableHeadless: true,
      locationAuthorizationAlert: {
        "titleWhenNotEnabled": strings.locationAuthzPermissionAlert,
        "titleWhenOff": strings.locationAuthzPermissionAlert,
        "instructions": strings.locationAuthzPermissionAlertDesc,
        "cancelButton": strings.cancel,
        "settingsButton": strings.settings
      },
      locationTemplate: '''{ "lat":<%= latitude %>, "lon":<%= longitude %>,
                "time":"<%= timestamp %>", "altitude":<%= altitude %>, "accuracy":<%= accuracy %>,
                "speed":<%= speed %>, "bearing":<%= heading %>, "activity":"<%= activity.type %>",
                "activityConfidence":<%= activity.confidence %> }''',
      geofenceTemplate: '''{ "lat":<%= latitude %>, "lon":<%= longitude %>,`
                "time":"<%= timestamp %>", "altitude":<%= altitude %>, "accuracy":<%= accuracy %>,
                "speed":<%= speed %>, "bearing":<%= heading %>, "activity":"<%= activity.type %>",
                "activityConfidence":<%= activity.confidence %> }''',
      notification: bg.Notification(
        title: strings.appName,
        text: strings.locationTrayNotification,
        smallIcon: "drawable/ic_notification",
        sticky: false,
        channelName: MoresConfig.geoNotificationChannel,
        priority: bg.Config.NOTIFICATION_PRIORITY_LOW,
      ),
      logLevel: bg.Config.LOG_LEVEL_INFO,
    )).then((bg.State state) {
      if (!state.enabled) {
        bg.BackgroundGeolocation.start();
        // bg.BackgroundGeolocation.changePace(true);
      }
    });
```

## Expected Behavior
Application do not crash when in background or terminated

## Actual Behavior
Applications crashes

## Context
Migrate application to flutter 2.x

## Debug logs
<details>
	<summary>Logs</summary>

``` <!-- syntax-highligting:  DO NOT REMOVE -->
04-01 07:44:38.781 21290 21290 D AndroidRuntime: Shutting down VM
04-01 07:44:38.782 21290 21290 E AndroidRuntime: FATAL EXCEPTION: main
04-01 07:44:38.782 21290 21290 E AndroidRuntime: Process: eu.mores, PID: 21290
04-01 07:44:38.782 21290 21290 E AndroidRuntime: java.lang.NullPointerException: Attempt to read from field 'java.lang.String io.flutter.embedding.engine.g.b.b' on a null object reference
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at io.flutter.embedding.engine.g.c.a(Unknown Source:2)
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at com.transistorsoft.flutter.backgroundgeolocation.HeadlessTask.startBackgroundIsolate(Unknown Source:20)
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at com.transistorsoft.flutter.backgroundgeolocation.HeadlessTask.dispatch(Unknown Source:4)
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at com.transistorsoft.flutter.backgroundgeolocation.HeadlessTask.run(Unknown Source:0)
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:883)
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:100)
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:237)
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:8167)
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
04-01 07:44:38.782 21290 21290 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
04-01 07:44:38.784 21290 21322 I TSLocationManager: [c.t.l.service.TrackingService a] 
04-01 07:44:38.784 21290 21322 I TSLocationManager:   ðµ  setPace: true â true
04-01 07:44:38.784 21290 21290 E TSLocationManager: [c.t.l.a.BackgroundGeolocation$v0 uncaughtException] 
04-01 07:44:38.784 21290 21290 E TSLocationManager:   âŒïž  Uncaught Exception: Attempt to read from field 'java.lang.String io.flutter.embedding.engine.g.b.b' on a null object reference
04-01 07:44:38.784 21290 21290 E TSLocationManager: {"activityRecognitionInterval":10000,"allowIdenticalLocations":false,"authorization":{},"autoSync":true,"autoSyncThreshold":0,"backgroundPermissionRationale":{"title":"Allow {applicationName} to access this device's location even when closed or not in use?","message":"[CHANGEME] This app collects location data for FEATURE X and FEATURE Y.","positiveAction":"Change to \"{backgroundPermissionOptionLabel}\"","negativeAction":""},"batchSync":false,"configUrl":"","debug":false,"deferTime":0,"desiredAccuracy":100,"desiredOdometerAccuracy":100,"disableAutoSyncOnCellular":false,"disableElasticity":false,"disableLocationAuthorizationAlert":false,"disableMotionActivityUpdates":false,"disableStopDetection":false,"distanceFilter":1000,"elasticityMultiplier":1,"enableHeadless":true,"enableTimestampMeta":false,"extras":{"instanceId":"b9aea12cb3a66a61"},"fastestLocationUpdateInterval":-1,"foregroundService":true,"geofenceInitialTriggerEntry":true,"geofenceModeHighAccuracy":false,"geofenceProximityRadius":1000,"geofenceTemplate":"{ \"lat\":<%= latitude %>, \"lon\":<%= longitude %>,`\n                \"time\":\"<%= timestamp %>\", \"altitude\":<%= altitude %>, \"accuracy\":<%= accuracy %>,\n                \"speed\":<%= speed %>, \"bearing\":<%= heading %>, \"activity\":\"<%= activity.type %>\",\n                \"activityConfidence\":<%= activity.confidence %> }","headers":{},"headlessJobService":"com.transistorsoft.flutter.backgroundgeolocation.HeadlessTask","heartbeatInterval":-1,"httpRootProperty":"location","httpTimeout":60000,"isMoving":true,"locationAuthorizationRequest":"Always","locationTemplate":"{ \"lat\":<%= latitude %>, \"lon\":<%= longitude %>,\n                \"time\":\"<%= timestamp %>\", \"altitude\":<%= altitude %>, \"accuracy\":<%= accuracy %>,\n                \"speed\":<%= speed %>, \"bearing\":<%= heading %>, \"activity\":\"<%= activity.type %>\",\n                \"activityConfidence\":<%= activity.confidence %> }","locationTimeout":60,"locationUpdateInterval":1000,"locationsOrderDirection":"ASC","logLevel":3,"logMaxDays":3,"maxBatchSize":5,"maxDaysToPersist":3,"maxRecordsToPersist":-1,"method":"POST","minimumActivityRecognitionConfidence":75,"motionTriggerDelay":0,"notification":{"layout":"","title":"ZisÅ¥ujem polohu","text":"Location Service activated","color":"","channelName":"geolocation","smallIcon":"drawable\/ic_notification","largeIcon":"","priority":-1,"sticky":false,"strings":{},"actions":[]},"params":{},"persist":true,"persistMode":2,"schedule":[],"scheduleUseAlarmManager":false,"speedJumpFilter":300,"startOnBoot":true,"stationaryRadius":25,"stopAfterElapsedMinutes":0,"stopOnStationary":false,"stopOnTerminate":false,"stopTimeout":1,"triggerActivities":"on_foot, walking, running","url":"https:\/\/api.dev.mores.eu\/public\/metrics\/user\/location","useSignificantChangesOnly":true,"enabled":true,"schedulerEnabled":false,"trackingMode":1,"odometer":17359.62109375,"isFirstBoot":false,"didLaunchInBackground":false}
04-01 07:44:38.784 21290 21290 E TSLocationManager: java.lang.NullPointerException: Attempt to read from field 'java.lang.String io.flutter.embedding.engine.g.b.b' on a null object reference
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at io.flutter.embedding.engine.g.c.a(Unknown Source:2)
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at com.transistorsoft.flutter.backgroundgeolocation.HeadlessTask.startBackgroundIsolate(Unknown Source:20)
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at com.transistorsoft.flutter.backgroundgeolocation.HeadlessTask.dispatch(Unknown Source:4)
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at com.transistorsoft.flutter.backgroundgeolocation.HeadlessTask.run(Unknown Source:0)
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at android.os.Handler.handleCallback(Handler.java:883)
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at android.os.Handler.dispatchMessage(Handler.java:100)
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at android.os.Looper.loop(Looper.java:237)
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at android.app.ActivityThread.main(ActivityThread.java:8167)
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at java.lang.reflect.Method.invoke(Native Method)
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
04-01 07:44:38.784 21290 21290 E TSLocationManager: 	at com.android.internal.os.ZygoteIni
```

</details>

**Comment by christocracy** [comment]
A similar issue was posted to [background_fetch](https://github.com/transistorsoft/flutter_background_fetch/issues/203#issuecomment-809556485).  Flutter 2 broke Android Headless.

It's fixed in `4.0.0-nullsafety.1`.

**Comment by hlavki** [comment]
Thanks for the prompt release, we can close it now and reopen it if problem occurs again.
